meshmode/dof_array.py

```diff
@@ -294,23 +294,44 @@ class _EntryNotProvided:
 
 
 @serialize_container.register(DOFArray)
-def _(ary: DOFArray):
+def _serialize_container_dofarray(ary: DOFArray):
     return enumerate(ary._data)
 
 
-@deserialize_container_class.register(DOFArray)
-def _(cls, template: Any, iterable: Iterable[Tuple[Any, Any]], *,
-        actx: Optional[ArrayContext] = None):
-    iterable = list(iterable)
-    result = [_EntryNotProvided] * len(iterable)
+def _raise_index_inconsistency(i, stream_i):
+    raise ValueError("out-of-sequence indices supplied in DOFArray deserialization "
+            f"(expected {i}, received {stream_i})")
```

Comment:
Local to `_deserialize_container_dofarray`?

Reply:
6063560